source/configuration_manual/mail_location/obox/amazon_s3.rst

```diff
@@ -35,35 +42,66 @@ partitions = this would theoretically provide you up to: ``~896,000
 PUT/POST/DELETE TPS and 1,408,000`` GET TPS if requests are distributed evenly
 across the partitions).
 
+Configuration
+^^^^^^^^^^^^^
+
 .. code-block:: none
 
   plugin {
-    obox_fs = fscache:1G:/var/cache/mails:compress:gz:6:s3: https://ACCESSKEY:SECRET@BUCKETNAME.s3.amazonaws.com/
-    obox_index_fs = compress:gz:6:s3: https://ACCESSKEY:SECRET@BUCKETNAME.s3.amazonaws.com/
+    obox_fs = fscache:1G:/var/cache/mails:compress:gz:6:aws-s3:https://ACCESSKEY:SECRET@BUCKETNAME.s3.amazonaws.com/
+    obox_index_fs = compress:gz:6:aws-s3:https://ACCESSKEY:SECRET@BUCKETNAME.s3.amazonaws.com/
   }
 
+.. Note::
+        If the version you are using does not support the ``aws-s3`` scheme,
+        try using the more generic ``s3`` scheme.
+
+        .. versionadded:: 2.3.10
+
 Get ACCESSKEY and SECRET from `www.aws.amazon.com <https://aws.amazon.com/>`_
 ``-> My account -> Security credentials -> Access`` credentials. Create the
 ``BUCKETNAME`` from ``AWS Management Console -> S3 -> Create Bucket``.
 
 If the ``ACCESSKEY`` or ``SECRET`` contains any special characters, they can be
 %hex-encoded.
 
+If the ``aws-s3`` scheme is used Dovecot defaults to prepend the following URL
+parameters, refer to :ref:`http_based_object_storages` for details. The same
+setup can be achieved by using the ``s3`` scheme and adding the parameters
+manually.
+
+.. code-block:: none
+
+  addhdrvar=x-amz-security-token:%{auth:token}&loghdr=x-amz-request-id&loghdr=x-amz-id-2
```

Comment:
Needs to mention auth_role setting in some way. That using it enables doing IAM lookups, which then overwrites the access key/secret.